modules/general/storage-account-rbac/main.bicep

```diff
@@ -0,0 +1,62 @@
+@description('The name of the storage account to apply the role assignement.')
+param storageAccountName string
+
+@allowed([
+  'Owner'
+  'Contributor'
+  'Reader'
+  'Storage Blob Data Owner'
+  'Storage Blob Data Contributor'
+  'Storage Blob Data Reader'
+])
+@description('The name of the role to grant the assignee on the storage account. See: https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles')
+param role string
+
+@description('The object ID of the assignee Azure AD group / service principal / user.')
+param assigneeObjectId string
+
+@allowed([
+  'Group'
+  'ServicePrincipal'
+  'User'
+])
+@description('The assignee\'s type of Azure AD principal.')
+param principalType string
+
+@description('A GUID to use as the role assignment resource name, if omitted, one will be generated based on the storage account, role and assignee')
+param roleAssignmentId string = ''
+
+
+targetScope = 'resourceGroup'
+
+
+// provide a more user-friendly roleDefinitionId lookup
+var roles = {
+  'Owner': resourceId('Microsoft.Authorization/roleDefinitions', '8e3af657-a8ff-443c-a75c-2fe8c4bcb635')
```

Comment:
What do you think about having an internal module that maintains these mappings and outputs a dictionary of all roles we end up referencing over time?  We can still validate the set of roles that are relevant to a particular scenario, but would avoid some repetition of those common roles.

Reply:
Added an issue for this https://github.com/endjin/Endjin.RecommendedPractices.Bicep/issues/56